src/account/eth_account/eth_account.cairo

```diff
@@ -0,0 +1,256 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts for Cairo vX.Y.Z (account/eth_account/eth_account.cairo)
+
+/// # EthAccount Component
+///
+/// The EthAccount component enables contracts to behave as accounts signing with Ethereum keys.
+#[starknet::component]
+mod EthAccountComponent {
+    use openzeppelin::account::eth_account::interface::EthPublicKey;
+    use openzeppelin::account::eth_account::interface;
+    use openzeppelin::account::utils::{execute_calls, is_valid_eth_signature};
+    use openzeppelin::introspection::src5::SRC5Component::InternalTrait as SRC5InternalTrait;
+    use openzeppelin::introspection::src5::SRC5Component;
+    use poseidon::poseidon_hash_span;
+    use starknet::account::Call;
+    use starknet::get_caller_address;
+    use starknet::get_contract_address;
+    use starknet::get_tx_info;
+
+
+    const TRANSACTION_VERSION: felt252 = 1;
+    // 2**128 + TRANSACTION_VERSION
+    const QUERY_VERSION: felt252 = 0x100000000000000000000000000000001;
+
+    #[storage]
+    struct Storage {
+        EthAccount_public_key: EthPublicKey
+    }
```

Comment:
The key is not compatible anyway, so this won't be a module that can be upgraded, I don't think is an issue because we didn't see usage of the cairo 0 module, but probably worth keeping in mind. cc @martriay

Reply:
If Account_public_key (vs account_public_key) is only there for storage BC, can we avoid caps here? e.g. eth_account_public_key, or just public_key?

Reply:
> If Account_public_key (vs account_public_key) is only there for storage BC, can we avoid caps here? e.g. eth_account_public_key, or just public_key?

We have the convention of adding the module name as a prefix, and the components and contracts modules are capitalized in the library. That would be a breaking change if we'd want to adopt them and update module naming conventions.